test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+				const account = yield db.one(
+					'SELECT encode("publicKey", \'hex\') as "publicKey" FROM mem_accounts LIMIT 1'
+				);
+				const rewards = yield db.blocks.aggregateBlocksReward({
+					generatorPublicKey: account.publicKey,
+					start: (+new Date('2017.01.01') / 1000).toFixed(0),
+					end: (+new Date() / 1000).toFixed(0),
+				});
+
+				expect(rewards).to.be.not.empty;
+				expect(rewards).to.be.an('array');
+				return expect(rewards[0]).to.have.all.keys(
+					'delegate',
+					'count',
+					'fees',
+					'rewards'
+				);
+			});
+		});
+
+		describe('count()', () => {
+			it('should use the correct SQL file with no parameters', function*() {
+				sinonSandbox.spy(db, 'one');
+				yield db.blocks.count();
+
+				expect(db.one.firstCall.args[0]).to.eql(sql.count);
+				expect(db.one.firstCall.args[1]).to.eql([]);
+				return expect(db.one).to.be.calledOnce;
+			});
+
+			it('should return integer type count of total blocks', function*() {
+				const count = yield db.blocks.count();
+
+				expect(count).to.be.an('number');
+				return expect(count).to.be.eql(5); // Number of blocks in seed (db_seed.js#23)
+			});
+		});
+
+		describe('list()', () => {
+			it('should use raw sql with provided parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+				const params = {
+					where: '',
+					sortField: '',
+					limit: 10,
+					offset: 0,
+				};
+				yield db.blocks.list(params);
+
+				expect(db.query.firstCall.args[0]).to.be.a('string');
+				expect(db.query.firstCall.args[1]).to.eql(params);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should be rejected with error if where param is not provided as array', () => {
+				return expect(
+					db.blocks.list({ where: "b_id = '1111'", offset: 0, limit: 1 })
+				).to.be.eventually.rejectedWith('Invalid parameter "where" provided.');
+			});
+
+			it('should be rejected with error if limit parameter is not specified', () => {
+				return expect(
+					db.blocks.list({ offset: 0 })
+				).to.be.eventually.rejectedWith("Property 'limit' doesn't exist.");
+			});
+
+			it('should be rejected with error if offset parameter is not specified', () => {
+				return expect(
+					db.blocks.list({ limit: 10 })
+				).to.be.eventually.rejectedWith("Property 'offset' doesn't exist.");
+			});
+
+			it('should return blocks with valid format', function*() {
+				const blocks = yield db.blocks.list({ limit: 1, offset: 0 });
+
+				expect(blocks).to.be.an('array');
+				expect(blocks).to.have.lengthOf(1);
+				return expect(blocks[0]).to.have.all.keys([
+					'b_blockSignature',
+					'b_confirmations',
+					'b_generatorPublicKey',
+					'b_height',
+					'b_id',
+					'b_numberOfTransactions',
+					'b_payloadHash',
+					'b_payloadLength',
+					'b_previousBlock',
+					'b_reward',
+					'b_timestamp',
+					'b_totalAmount',
+					'b_totalFee',
+					'b_version',
+				]);
+			});
+
+			it('should paginate the blocks properly', function*() {
```

Comment:
fragile, sort by unique field should be used as we cannot expect postgres to always sort same way
but maybe for that small data set can be ok...